src/client/TabList/index.tsx

```diff
@@ -91,6 +91,13 @@ export const TabList = ({
       );
     };
   }, [handleKeyPress]);
+  //Checks the current state of the tablist and if it is an invalid file it removes it from the files array.
+  for (let i = 0; i < tabList.length; i++) {
+    if (files[tabList[i]] == undefined) {
+      closeTab(tabList[i]);
+      tabList = tabList.splice(i, i);
```

Comment:
Thank you for this attempt! My gut feeling is that this is not the ideal place to do this adjustment. The splicing logic is good, but I feel like this should ideally be done elsewhere. This solution here mutates the `tabList` locally, but ideally the _definition_ of `tabList` should change so that the file being deleted is removed from there, then the file could be actually removed from the document and deleted.

The case we are solving for is when you delete a file _and_ that same file is currently open in a tab. Ideally the logic that deletes the file would trickle down in an elegant way to update the tab list.